aziotctl/aziotctl-common/src/system_logs.rs

```diff
@@ -0,0 +1,17 @@
+use std::process::Command;
+
+#[allow(clippy::module_name_repetitions)]
+pub fn get_system_logs(processes: &[&str], mut additional_args: &[&str]) {
+    let processes = processes.iter().flat_map(|p| vec!["-u", p]);
```

Comment:
`&["-u", p]` instead of allocating a new `Vec` for each iterator item. Plus a `.copied()` at the end to convert the `Iterator<Item = &&str>` to `Iterator<Item = &str>` as `Command::args` wants.

Reply:
I used a vec because using a slice gives:
```
error[E0515]: cannot return reference to temporary value
 --> aziotctl/aziotctl-common/src/system_logs.rs:7:51
  |
7 |     let processes = processes.iter().flat_map(|p| &["-u", p]);
  |                                                   ^---------
  |                                                   ||
  |                                                   |temporary value created here
  |                                                   returns a reference to data owned by the current function

error: aborting due to previous error
```

Reply:
Ah, right, the slice is still a temporary even if the borrows are longer-lived.

Then do `std::iter::once("-u").chain(std::iter::once(p))` instead. And that won't need a `.copied()` at the end.

Reply:
I'll also throw out another alternative: `Some("-u").into_iter().chain(Some(p).into_iter())`. I find `std::iter::once` kinda redundant, since `Option` is always in-scope, and implements `IntoIter`.

 `Some("-u").into_iter().chain(Some(p).into_iter())`.

Reply:
Honestly, I think the vec is much more readable, and I'm not too concerned about 4 extra allocations of a few bytes. This is a command line tool that is run rarely, I don't think it needs to be optimized.

Reply:
Yeah, I'm with Lee on this one. It's fun to optimize these sorts of things, but in this case it doesn't really matter, and `vec!` is far more readable.